devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -313,6 +313,17 @@ def zwaveEvent(DoorLockOperationReport cmd) {
 		}
 	}
 	result ? [createEvent(map), *result] : createEvent(map)
+	if (generatesDoorLockOperationReportBeforeAlarmReport()) {
+		// we're expecting lock events to come after notification events, but for specific yale locks they come out of order
+		runIn(3, "delayLockEvent", [data: [event: result]])
```

Comment:
@greens, @mckeed: Does this work? I see a blunder:

`result` will not have any data for locked/unlocked event as you are not putting that into the `result` list. So the changes will not work when expected to work. Besides this, the changes will also break the existing functionality of `refresh` in DTH.

Reply:
Ah thanks, didn't notice there wasn't actually an assignment on the line above.